### PR TITLE
Docker Login: support with or without email

### PR DESCRIPTION
**Pull request by nathanleiby**
Tries to login with email. Falls back to no email.

Tested on CircleCI v1 and v2 (different docker versions with diff behavior) and both succeed.

### CODE REVIEW

File: circleci/docker-publish [hunk 1]
```diff
@@ -15,7 +15,6 @@ check_hub_vars() {
   # User supplied args
   if [[ -z $DOCKER_USER ]]; then echo "Missing arg1 DOCKER_USER" && exit 1; fi
   if [[ -z $DOCKER_PASS ]]; then echo "Missing arg2 DOCKER_PASS" && exit 1; fi
-  if [[ -z $DOCKER_EMAIL ]]; then echo "Missing arg3 DOCKER_EMAIL" && exit 1; fi
   if [[ -z $ORG ]]; then echo "Missing arg4 ORG" && exit 1; fi
```
Comment (burnsed) [inline circleci/docker-publish]: is now arg3? or do we need to handle either case?

File: circleci/docker-publish [hunk 2]
```diff
@@ -44,9 +46,11 @@ SHORT_SHA=${CIRCLE_SHA1:0:7}
 
 DOCKER_USER=$1
 DOCKER_PASS=$2
-DOCKER_EMAIL=$3
 ORG=$4
```
Comment (burnsed) [inline circleci/docker-publish]: $3 ?

Comment (nathanleiby) [reply circleci/docker-publish]: During this change, I don't plan on updating all the other repos (I'd have to update their calls to remove docker email) ... So I'm just ignoring that arg

File: circleci/docker-publish (hunk 1 above)
Comment (nathanleiby) [reply circleci/docker-publish]: Ah yeah, I think we should still keep this validation logic, even though we don't use it

File: circleci/docker-publish (hunk 2 above)
Comment (burnsed) [reply circleci/docker-publish]: Oh I see now. makes sense

**Pull request merged**